src/components/external-source/ExternalSourceManager.svelte

```diff
@@ -0,0 +1,897 @@
+        >
+          <XIcon />
+        </button>
+      {/if}
+    </svelte:fragment>
+
+    <svelte:fragment slot="body">
+      {#if selectedEvent}
+        <ExternalEventForm externalEvent={selectedEvent} showHeader={true} />
+      {:else if selectedSource}
+        <div class="external-source-header">
+          <div class="external-source-header-title">
+            <div class="external-source-header-title-value st-typography-medium">
+              {selectedSource.key}
+            </div>
+          </div>
+        </div>
+        <div class="selected-source-forms">
+          <fieldset>
+            <Input layout="inline">
+              Source Type
+              <input
+                class="st-input w-100"
+                disabled={true}
+                name="source-type"
+                value={selectedSource.source_type_name}
+              />
+            </Input>
+
+            <Input layout="inline">
+              Derivation Group
+              <input
+                class="st-input w-100"
+                disabled={true}
+                name="derivation-group"
+                value={selectedSource.derivation_group_name}
+              />
+            </Input>
+
+            <Input layout="inline">
+              Owner
+              <input class="st-input w-100" disabled={true} name="owner" value={selectedSource.owner} />
+            </Input>
+
+            <Input layout="inline">
+              {`Start Time (${$plugins.time.primary.label})`}
+              <DatePicker
+                dateString={formatDate(new Date(selectedSource.start_time), $plugins.time.primary.format)}
+                disabled={true}
+                name="start-time"
+              />
+            </Input>
+
+            <Input layout="inline">
+              {`End Time (${$plugins.time.primary.label})`}
+              <DatePicker
+                dateString={formatDate(new Date(selectedSource.end_time), $plugins.time.primary.format)}
+                disabled={true}
+                name="end-time"
+              />
+            </Input>
+
+            <Input layout="inline">
+              {`Valid At (${$plugins.time.primary.label})`}
+              <DatePicker
+                dateString={formatDate(new Date(selectedSource.valid_at), $plugins.time.primary.format)}
+                disabled={true}
+                name="valid-at"
+              />
+            </Input>
+
+            <Input layout="inline">
+              {`Created At (${$plugins.time.primary.label})`}
+              <DatePicker
+                dateString={formatDate(new Date(selectedSource.created_at), $plugins.time.primary.format)}
+                disabled={true}
+                name="valid-at"
+              />
+            </Input>
+
+            <Collapse defaultExpanded={false} title="Event Types" tooltipContent="View Contained Event Types">
+              {#await effects.getExternalEventTypesBySource(selectedSource.key, selectedSource.derivation_group_name, user)}
```

Comment:
Can we avoid this pattern of inline awaiting? This is a cool svelte feature, but it's not our current convention and we potentially loose control of the request if we decide we need to abort the request when `selectedSource` changes mid request.

Reply:
Updated these to set `selectedSourceEventTypes` and `selectedSourceMetadata` inside of `selectSource` - because we don't currently track `selectedSource` as a writable store I didn't see an easy way to make a derived store for these, but that's something we could do if its preferred to this solution!

Reply:
Yup! This is a perfectly reasonable implementation!